pkgs/applications/misc/josm/default.nix

```diff
@@ -0,0 +1,31 @@
+{ fetchurl, stdenv, bash, jre8 }:
+
+stdenv.mkDerivation rec {
+  name = "josm-${version}";
+  version = "9060";
+
+  src = fetchurl {
+    url = "https://josm.openstreetmap.de/download/josm-snapshot-${version}.jar";
+    sha256 = "0c1q0bs3x1j9wzmb52xnppdyvni4li5khbfja7axn2ml09hqa0j2";
+  };
+
+  phases = [ "installPhase" ];
+
+  buildInputs = [ jre8 ];
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cat > $out/bin/josm <<EOF
+    #!${bash}/bin/bash
+    exec ${jre8}/bin/java -jar $src "\$@"
```

Comment:
I'm not sure if referencing `$src` from an installed file is very clean (makes it hard to find the jar, as you'd expect it in `$out`). Maybe copy the jar to `$out/share/josm`?